tools/libboard/board_dc/board_dc.c

```diff
@@ -326,3 +340,126 @@ fpga_result print_board_info(fpga_token token)
 
 	return resval;
 }
+
+// Read sysfs
+fpga_result read_sysfs(fpga_token token, char *sysfs_path, char *sysfs_name)
+{
+	fpga_result res = FPGA_OK;
+	fpga_result resval = FPGA_OK;
+	uint32_t size = 0;
+	char name[FPGA_STR_SIZE] = { 0 };
+	fpga_object sec_object;
+
+	if (sysfs_path == NULL ||
+		sysfs_name == NULL) {
+		OPAE_ERR("Invalid input parameter");
+		return FPGA_INVALID_PARAM;
+	}
+
+	res = fpgaTokenGetObject(token, sysfs_path, &sec_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get token Object");
+		return res;
+	}
+
+	res = fpgaObjectGetSize(sec_object, &size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to get object size ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	if (size > FPGA_STR_SIZE) {
+		OPAE_ERR("object size bigger then buffer size");
+		resval = FPGA_EXCEPTION;
+		goto out_destroy;
+	}
+
+	res = fpgaObjectRead(sec_object, (uint8_t *)(&name), 0, size, 0);
+	if (res != FPGA_OK) {
+		OPAE_ERR("Failed to Read object ");
+		resval = res;
+		goto out_destroy;
+	}
+
+	snprintf(sysfs_name, FPGA_STR_SIZE, "%s", (char *)name);
```

Comment:
This is using snprintf( ) to perform a string copy.  It should probably use strnlen( ), memcpy( ), and an explicit null termination instead.

Reply:
fixed